manager/integration/tests/test_upgrade.py

```diff
@@ -31,53 +53,177 @@ def longhorn_upgrade(image_tag):
 
 
 @pytest.mark.upgrade
-def test_upgrade(upgrade_image_tag, volume_name):  # NOQA
+def test_upgrade(upgrade_image_tag, volume_name, pod_make, statefulset, storage_class):  # NOQA
     """
     Test Longhorn upgrade
 
+    Prerequisite:
+      - Disable Auto Salvage Setting
```

Comment:
Does this test case has a side-effect of disabling the auto salvage setting?

Reply:
added `reset_settings` to the end of the test.

Reply:
That won't work if the test failed.